[Jared Lim] iP

jared98lyj opened the pull request.

jared98lyj committed:
Added the driver function to take in user input.
Added ArrayList to store input.

jared98lyj committed:
Level 2

jared98lyj committed:
Updated to level 3.

jared98lyj committed:
Level-4
Improved layout of code.
Added children classes for task to give more granularity to the different types of tasks.

jared98lyj committed:
Level-5
Added try-catch blocks to flag errors, and prevent possible incorrect input from created unwanted stops/breakages in the program.

jared98lyj committed:
Level-6
Added the removeTask method for deletion

jared98lyj committed:
Renamed to Duke.java

justgnohUG commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,135 @@
+
+    /**
+     * Main class to take in user input.
+     * @param args Filler
```
Perhaps a more meaningful name for @param could be possibly replaced here

justgnohUG commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    /**
+     * Overriden constructor to account for deadline and events that have timestamps.
```
Not too sure if "Overriden" is the appropriate term here. I think "Overloaded constructor" may be more appropriate.

wei-yutong replied:
i think the term you meant to use might be "overloaded" instead

justgnohUG commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,135 @@
+    }
+
+    /**
+     * Enumerates all tasks in the lis using 1-based indexing.
```
Minor typo for "list"

justgnohUG commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,55 @@
+     * @param description Description of method
+     * @param eventDate Boolean flag indicating whether task is done
+     */
+    public Task(String description, String eventDate) {
```
Might be a good idea to add "@Override" tag if this method is truly overriden

wei-yutong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,135 @@
+/**
+ * Driver class that takes in input and performs certain functions based on input.
+ */
 public class Duke {
```
this class is rather long, perhaps you could consider abstracting the methods out?

wei-yutong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,135 @@
+            } else if (command.equals("list")) {
+                enumerateTasks();
+            } else if (command.startsWith("done")) {
+                String[] delString = command.split("\\s+");
```
delString is not a very intuitive variable name, maybe you could make it more specific?

wei-yutong commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,55 @@
+    protected boolean isDone;
+
+    enum Level {
+        low,
+        average,
+        high
```
I think that constants in enum should be in all uppercase, eg "LOW", "AVERAGE" and "HIGH"

wei-yutong commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    /**
+     * Overriden method to mark a task as done.
```
Did you mean "Overloaded" here as well? I've noticed it in a few other places. Otherwise, as the other reviewer mentioned, overridden methods should tagged with "@Override"

jared98lyj committed:
Level-7
Added I/O utility.
Autocreate data directory if it does not exist.

wei-yutong reviewed (commented):
your code is generally very clean and easy to read! i liked how your method names were clear and how you separated long lines accurately :)

jared98lyj committed:
Level-8
Added date time formatter across all task classes. Created a ParseDate class to handle date formatting.

jared98lyj committed:
A-J-Unit
Add-gradle-support
Included J-unit tests under dependencies in build.gradle file
Added gradle wrapper files as well

jared98lyj committed:
Level-9
Added a find function uitlity to filter tasks by keyword

jared98lyj committed:
A-MoreOOP
Refactored a bulk of the code from Duke.java into four classes: Parser, Storage, TaskList, and Ui.
Modified Duke.java to accomodate these changes as well.

jared98lyj committed:
Add JUnit Tests
Included the gradle wrapper and the associated dependencies for the Junit test.
Added two JUnit Test: One to test the loading of the stored text file, the other to test marking as done utility. Both under src/test/java.

jared98lyj committed:
Made the classes more OOP-like.
Applied changes to all other dependent classes as well.

jared98lyj committed:
Level-10
Added GUI and modified the corresponding methods to return a string (rather than stdout) to feed into the user input functions, which are returned and displayed in the dialogue box in GUI.

jared98lyj committed:
Added important assertions throughout the code. Some of them include checks for non empty objects, and list to avoid unchecked null pointers which can crash the program.

jared98lyj committed:
Improved abstration in the Duke.java, erasing direct array/list access as much as possible by abstracting it away into the Task classes.

jared98lyj committed:
Improved the find function to be more flexible, giving the user the option to specify type of search: partial case sensitive, partial non- case sensitive, full exact match.

jared98lyj committed:
Built a basic GUI for interaction betwen user and bot.
GUI echoes user commands, returns output by bot.

jared98lyj committed:
Added a simple user guide with some usage examples and corresponding output

jared98lyj committed:
Added java doc comments for most methods, less the getter and setter methods which are quite clear.

jared98lyj committed:
Improved code quality by ensuring similar levels of abstraction in a method body. Prevented as much as possible the use of magic strings when it comes to important kinds of data. Avoided direct usage of list indexing in Duke method by
abstracting to the TaskList method.

jared98lyj committed:
Refactored CLI version of Duke.java.

jared98lyj committed:
Improved consistency of String representation and formatting.

jared98lyj committed:
Added input and expected text to the files. Debugged some of the inconsistencies for specific inputs like missing spaces.

jared98lyj committed:
Fixed launching issues with duke.jar file.

jared98lyj committed:
Made final changes to the readMe file and added a background image.

jared98lyj committed:
Update README.md

jared98lyj committed:
Delete README.md

jared98lyj committed:
Updated minor changes to UI class.

jared98lyj committed:
Set theme jekyll-theme-slate

jared98lyj committed:
Commtting the Ui.png file to docs directory

jared98lyj committed:
Revert "Commtting the Ui.png file to docs directory"

This reverts commit 518584ef8e73fb8b4c961af9e86ca84237a750a7.

jared98lyj committed:
Merge branch 'master' of https://github.com/jared98lyj/ip

jared98lyj committed:
Commiting the Ui.png to docs directory

jared98lyj committed:
no message

jared98lyj committed:
Renamed image file to Ui.png

jared98lyj committed:
Renamed to Ui.png

jared98lyj committed:
Update README.md

jared98lyj committed:
Update README.md
